components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

```diff
@@ -36,3 +39,17 @@ fun String.isPhone(): Boolean = contains("tel:", true)
 fun String.isEmail(): Boolean = contains("mailto:", true)
 
 fun String.isGeoLocation(): Boolean = contains("geo:", true)
+
+/**
+ * Converts a [String] to a [Date] object, if an empty string is given will return Date().
+ * @property format of the this given [String] object.
+ * @property locale in which you want to format this [String] object.
```

Comment:
`@param` ... locale the locale to use when converting the String, defaults to ....